hack/tools/Makefile

```diff
@@ -74,7 +75,7 @@ $(GINKGO): go.mod
 	mkdir -p $(BIN_DIR)
 	go build -tags=tools -o $(BIN_DIR)/ginkgo github.com/onsi/ginkgo/ginkgo
 
-COUNTERFEITER := $(BIN_DIR)/counterfeiter
+counterfeiter: $(COUNTERFEITER)
 $(COUNTERFEITER): $(BIN_DIR) go.mod go.sum
```

Comment:
I realize you didn't change this but I wonder why there is a dependency on `$(BIN_DIR)`?  It may look reasonable in case the whole directory is removed, but in practice it actually breaks things in that case because there is no target for `$(BIN_DIR)`.

```
$ cd hack/tools;make clean;cd -
rm -rf bin
~git/tanzu-framework
$ make hack/tools/bin/counterfeiter
make -C /Users/kmarc/git/tanzu-framework/hack/tools bin/counterfeiter
make[1]: *** No rule to make target `bin', needed by `bin/counterfeiter'.  Stop.
make: *** [hack/tools/bin/counterfeiter] Error 2
```
but if removing that dependency, it will pass because the `go build` will create the directory.

Just a thought 😄